exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -436,143 +542,329 @@ func (e *Exchange) GetWithdrawalsHistory(ctx context.Context, c currency.Code, _
 
 // GetRecentTrades returns the most recent trades for a currency and asset
 func (e *Exchange) GetRecentTrades(ctx context.Context, pair currency.Pair, assetType asset.Item) ([]trade.Data, error) {
-	return e.GetHistoricTrades(ctx, pair, assetType, time.Now().Add(-time.Minute*15), time.Now())
+	fPair, err := e.FormatExchangeCurrency(pair, assetType)
+	if err != nil {
+		return nil, err
+	}
+
+	var resp []trade.Data
+	switch assetType {
+	case asset.Spot:
+		tradeData, err := e.GetTrades(ctx, fPair, 0)
+		if err != nil {
+			return nil, err
+		}
+		var side order.Side
+		for i := range tradeData {
+			side, err = order.StringToOrderSide(tradeData[i].TakerSide)
+			if err != nil {
+				return nil, err
+			}
+			resp = append(resp, trade.Data{
+				Exchange:     e.Name,
+				CurrencyPair: fPair,
+				AssetType:    assetType,
+				Side:         side,
+				Price:        tradeData[i].Price.Float64(),
+				Amount:       tradeData[i].Amount.Float64(),
+				Timestamp:    tradeData[i].Timestamp.Time(),
+			})
+		}
+	case asset.Futures:
+		var tradeData []*FuturesExecutionInfo
+		tradeData, err = e.GetFuturesExecution(ctx, fPair.String(), 0)
+		if err != nil {
+			return nil, err
+		}
+		for i := range tradeData {
+			var side order.Side
+			side, err = order.StringToOrderSide(tradeData[i].Side)
+			if err != nil {
+				return nil, err
+			}
+			resp = append(resp, trade.Data{
+				Exchange:     e.Name,
+				CurrencyPair: fPair,
+				AssetType:    assetType,
+				Side:         side,
+				Price:        tradeData[i].Price.Float64(),
+				Amount:       tradeData[i].Amount.Float64(),
+				Timestamp:    tradeData[i].CreationTime.Time(),
+			})
+		}
+	default:
+		return nil, fmt.Errorf("%w asset type: %v", asset.ErrNotSupported, assetType)
```

Comment:
🚧 Change request:
🌴 **Please find all occurrences of this pattern in this PR.**

```suggestion
		return nil, fmt.Errorf("%w asset type: %v", asset.ErrNotSupported, assetType)
```